tests/foreman/longrun/test_provisioning_computeresource.py

```diff
@@ -0,0 +1,273 @@
+                      "storage_domain={0},"
+                      "bootable=True".format(self.storage_id),
+            'provision-method': 'build'
+        })
+        hostname = '{0}.{1}'.format(
+                host_name, self.config_env['domain'])
+        self.assertEquals(hostname, host['name'])
+        host_info = Host.info({'name': hostname})
+        host_ip = host_info.get('network').get('ipv4-address')
+        # Check on RHV, if VM exists
+        self.assertTrue(self.rhv_api.does_vm_exist(hostname))
+        # Get the information of created VM
+        rhv_vm = self.rhv_api.get_vm(hostname)
+        # Assert of Satellite mac address for VM and Mac of VM created is same
+        self.assertEqual(host_info.get('network').get('mac'),
+                         rhv_vm.get_nics()[0].mac
+                         )
+        # Start to run a ping check if network was established on VM
+        with self.assertNotRaises(ProvisioningCheckError):
+            host_provisioning_check(ip_addr=host_ip)
+
+    @run_only_on('sat')
+    @tier3
+    def test_positive_provision_vmware_with_host_group(self):
+        """ Provision a host on vmware compute resource with
+        the help of hostgroup.
+
+        :Requirement: Computeresource Vmware
+
+        :id: ae4d5949-f0e6-44ca-93b6-c5241a02b64b
+
+        :setup:
+
+            1. Vaild vmware hostname ,credentials.
+            2. Configure provisioning setup.
+            3. Configure host group setup.
+
+        :steps:
+
+            1. Go to "Hosts --> New host".
+            2. Assign the host group to the host.
+            3. Select the Deploy on as vmware Compute Resource.
+            4. Provision the host.
+
+        :expectedresults: The host should be provisioned with host group
+
+        :Caseautomation: Automated
+
+        :CaseLevel: System
+        """
+        cr_name = gen_string('alpha')
+        vmware_cr = make_compute_resource({
+            'name': cr_name,
+            'organizations': self.org_name,
+            'locations': self.loc_name,
+            'provider': FOREMAN_PROVIDERS['vmware'],
+            'server': self.vmware_server,
+            'user': self.vmware_username,
+            'password': self.vmware_password,
+            'datacenter': self.vmware_datacenter
+        })
+        self.assertEquals(vmware_cr['name'], cr_name)
+        host_name = gen_string('alpha').lower()
+        host = make_host({
+            'name': '{0}'.format(host_name),
+            'root-password': gen_string('alpha'),
+            'organization': self.org_name,
+            'location': self.loc_name,
+            'hostgroup': self.config_env['host_group'],
+            'pxe-loader': 'PXELinux BIOS',
+            'compute-resource-id': vmware_cr.get('id'),
+            'compute-attributes': "cpus=2,"
+                                  "corespersocket=2,"
+                                  "memory_mb=4028,"
+                                  "cluster=Satellite_Engineering,"
```

Comment:
do you think reading cluster, datastore values should be defined out of tests, may be in setup? For now its fine as one test is using them. But later if multiple tests uses them then reading from one place would be good.